ingress/loadbalancer.go

```diff
@@ -17,19 +37,181 @@ type LoadBalancerEntry struct {
 	ServicePort int32
 }
 
-type noopLoadBalancer struct {
+// Nginx implementation
+type nginxLoadBalancer struct {
+	binary    string
+	confDir   string
+	port      int
+	cmd       *exec.Cmd
+	signaller Signaller
 }
 
-func (lb *noopLoadBalancer) Update(entries []LoadBalancerEntry) error {
-	log.Debugf("Updating loadbalancer for %v", entries)
-	return nil
+// Used for generating nginx cofnig
+type loadBalancerTemplate struct {
+	Port       int
+	Entries    []LoadBalancerEntry
+	WorkingDir string
+}
+
+// DefaultSignaller sends os signals
+type DefaultSignaller struct {
+}
+
+// Sigquit sends a SIGQUIT to the process
+func (s *DefaultSignaller) Sigquit(p *os.Process) error {
+	return p.Signal(syscall.SIGQUIT)
+}
+
+// Sighup sends a SIGHUP to the process
+func (s *DefaultSignaller) Sighup(p *os.Process) error {
+	return p.Signal(syscall.SIGHUP)
 }
 
 // NewLB creates a new LoadBalancer
-func NewLB() LoadBalancer {
-	return &noopLoadBalancer{}
+func NewLB(nginxLocation, nginxConfDir string, port int, signaller Signaller) LoadBalancer {
```

Comment:
I think the signaller can be created internally - no need to require the controller to create it.